File: mozn-dashboard/server/internal/model/history.go
package model

// AlertSeverity is the severity of a historical alert.
type AlertSeverity string

const (
	SeverityCritical AlertSeverity = "critical"
	SeverityWarning  AlertSeverity = "warning"
	SeverityWatch    AlertSeverity = "watch"
	SeverityAdvisory AlertSeverity = "advisory"
)

// AlertOutcome is how a historical alert resolved.
type AlertOutcome string

const (
	OutcomeAllClear    AlertOutcome = "all-clear"
	OutcomeAutoCleared AlertOutcome = "auto-cleared"
)

// AlertHistoryRow is one row of the alert history table.
type AlertHistoryRow struct {
	ID       string        `json:"id"`
	Date     string        `json:"date"`
	Time     string        `json:"time"`
	Severity AlertSeverity `json:"severity"`
	Alert    string        `json:"alert"`
	Region   string        `json:"region"`
	AckTime  string        `json:"ackTime"`
	Duration string        `json:"duration"`
	Outcome  AlertOutcome  `json:"outcome"`
}

// AlertHistoryPage is the payload for the Alert History tab.
type AlertHistoryPage struct {
	Ranges     []string          `json:"ranges"`
	Regions    []string          `json:"regions"`
	Types      []string          `json:"types"`
	Severities []string          `json:"severities"`
	Rows       []AlertHistoryRow `json:"rows"`
}

File: mozn-dashboard/web/types/history.ts
/**
 * Alert History API contract types. Mirror the JSON served by the Go backend
 * (internal/model/history.go). Part of the shared types/ contract layer so the
 * fetch layer (lib/api) can reference them without reaching into features/. The
 * history feature re-exports these from features/history/types.
 */
import type { AlertSeverity, AlertOutcome } from "@/types/shared";

export type { AlertSeverity, AlertOutcome };

export interface AlertHistoryRow {
  id: string;
  date: string;
  time: string;
  severity: AlertSeverity;
  /** English "<param> — <station>", kept for search / type-facet / sort / CSV export. */
  alert: string;
  /** English parameter label — translatable via td(). */
  param: string;
  /** Station name (English) + Arabic counterpart, composed in the view by locale. */
  station: string;
  stationAr?: string;
  region: string;
  ackTime: string;
  duration: string;
  outcome: AlertOutcome;
}

export interface AlertHistoryPage {
  ranges: string[];
  regions: string[];
  types: string[];
  severities: string[];
  rows: AlertHistoryRow[];
}
